src/middlewares/verifyToken.js

```diff
@@ -0,0 +1,20 @@
+import JWT from 'jsonwebtoken';
+import { errorResponse } from '../utils';
+
+export default async (req, res, next) => {
+  try {
+    const { headers: { authorization } } = req;
+    const token = authorization.split(' ')[1];
+    if (!token || token === '') {
+      return errorResponse(res, 401, 'Access denied.');
+    }
+    const { userId } = await JWT.verify(token, process.env.SECRET);
+    if (!userId) {
+      return errorResponse(res, 401, 'Access denied.');
+    }
+    req.userId = userId;
+    return next();
+  } catch (error) {
+    return errorResponse(res, 500, 'We could not verify your identity. Please try again later');
```

Comment:
Okay. Cool.

Reply:
On a closer look, the error here isn't because of an invalid token, it is a server error. So the problem envisaged here isn't about the token. It's an internal error from the server, hence this response. If there's a better way I can phrase this, let me know. Thanks.

Reply:
The only most likely error to occur to be caught by that `catch` block will be the one thrown by `Jwt.verifyToken` function. Which would be because you are trying to verify an invalid token. The function does only one thing and that is token verification. Which is the only error that will be cause in that catch `block`. There is no database call in there or any other external call.

Reply:
OKay. Now I get. You're right. I guess I was thinking in terms of failures beyond the code. Like Network calls etc.